### PR TITLE
Handle authentication and user state on Ember front-end side

### PR DESCRIPTION
**Pull request by mklucsarits**

### CODE REVIEW

File: front/scripts/main/User.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		var userId = this.get('userId');
+		if (userId !== null) {
+			App.UserModel.find({userId: userId}).then((result: any) => {
+				this.set('model', result);
+			}, function () {});
+		}
+		this._super();
+	},
+
+	isAuthenticated: function (): boolean {
```
Comment (hakubo) [inline front/scripts/main/User.ts]: Ember.computed.bool

File: front/templates/main/components/site-head.hbs [hunk 2]
```diff
@@ -1,7 +1,13 @@
 <div {{action 'expandSideNav'}} class='side-nav-toggle'>
     {{svg 'nav' viewBox='0 0 24 14' class='icon nav'}}
 </div>
-{{render 'login'}}
+{{#if user.isAuthenticated}}
+	<figure class='avatar'>
+		<img {{bind-attr src=user.model.avatarPath}} {{bind-attr alt=user.model.name}}>
```
Comment (hakubo) [inline front/templates/main/components/site-head.hbs]:
```hbs
<img src={{user.model.avatarPath}} alt={{user.model.name}}>
```

File: front/scripts/main/User.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/// <reference path="./app.ts" />
+/// <reference path="./models/UserModel.ts" />
+'use strict';
+
+App.User = Em.Object.extend({
```
Comment (hakubo) [inline front/scripts/main/User.ts]: Why do we have User and UserModel separately?

Can't User just be model?

File: front/scripts/main/User.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	model: null,
+
+	init: function () {
+		var userId = this.get('userId');
+		if (userId !== null) {
+			App.UserModel.find({userId: userId}).then((result: any) => {
+				this.set('model', result);
+			}, function () {});
```
Comment (hakubo) [inline front/scripts/main/User.ts]: You implemented error handling in UserModel but here you don't use it. Can you explain what is the reason?

File: front/scripts/main/User.ts (hunk 3 above)
Comment (mklucsarits) [inline front/scripts/main/User.ts]: The reasoning was that the User object is for the current user only, while UserModel could be for any user, and utilized for such cases as profiles, messaging, etc. -- any place user data is needed.

Comment (hakubo) [inline front/scripts/main/User.ts]: so how about
```
App.CurrentUser
```

File: front/scripts/main/User.ts (hunk 4 above)
Comment (mklucsarits) [inline front/scripts/main/User.ts]: I think I meant to add something here but never did. I wasn't sure what should happen if the user data wasn't retrieved as expected. Any ideas @kenkouot ?

**Comment by mklucsarits** [comment]
This branch needs to be merged into the dev branch now, so I'm closing this PR.

**Pull request closed without merging**